### DISCUSSION TITLE
Containers on codespaces with docker-compose not resolving each other causing bootstrap to fail

### DISCUSSION THREAD

**Discussion by modamod** [community/community#37130] [Codespaces] [answered]
I am working on using codespaces for a laravel based project and I wanted to give it try using [snipe-it project](https://github.com/snipe/snipe-it). I was able to get it to work manually and I wanted to automate it using docker-compose, as I need to have multiple services ineracting with eachother. but the issue is that I got it to work locally using devconainers but when pushed to the repo and used with codespaces I get connection failure between the bootstrap container and the database when performing migrations.

This is snippet of the creation log with the DNS resolution error
[codespaces_error.log](https://github.com/community/community/files/9853190/codespaces_error.log)

This is the [link](https://github.com/hellomainland/snipe-it/tree/codespaces) to my repo with devcontainers configuration  on codespaces branch
Did anyone run into this before and how did you solve it?

**Comment by cmbrose** [comment, marked as answer]
Hi @modamod - thanks for reaching out and thanks for providing a lot of context!

I took your sample repo and trimmed it down to a small repro, it looks like the error is caused by a couple issues in the docker-compose setup.

The first thing I noticed was that the main `cli` container does not have a `depends_on` for the `mariadb` container, so it is not properly waiting for that container to set up. Additionally, once I added that I started getting errors for port collisions in the `mariadb` container and so I removed the `ports` setting in the docker-compose.yaml as well as the `EXPOSES` in the Dockerfile. 

From there it's still not _quite_ working, but I get these logs (`blankie` is the name of the repo I used for the smaller repro):

```
Running the postStartCommand from devcontainer.json...

Waiting for mariadb:3306.
Connected!
ERROR 1130 (HY000): Host 'cli.blankie_devcontainer_default' is not allowed to connect to this MariaDB server
ERROR 1130 (HY000): Host 'cli.blankie_devcontainer_default' is not allowed to connect to this MariaDB server
RESTORING
/workspaces/blankie/.devcontainer/bash/startup.sh: line 16: /workspaces/blankie/.gp/data/dump.sql: No such file or directory
Composer could not find a composer.json file in /workspaces/blankie
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
Could not open input file: artisan
Could not open input file: artisan
[3013 ms] postStartCommand failed with exit code 1. Skipping any further user-provided commands.
Finished configuring codespace.
```

It looks like there may be an additional setting needed in the `mariadb` setup to give the `cli` container access - it might be something you already have fixed in your larger scenario that I just didn't have in my small repro.

Hope this helps!

**Reply by modamod** [reply]
Thank you for looking into this issue. I will try to replicate the same config and see how it goes.

**Reply by cmbrose** [reply]
I published the repro here so you can see the configs I was working with: https://github.com/cmbrose/blankie/tree/snipe-it-compose

**Reply by modamod** [reply]
@cmbrose I managed to get pass the DNS resolution issues with your suggestions, thanks a lot. Now I have to deal with the reason why MariaDB refuses connections but I think I will be able to handle that. Again thanks alot.

**Reply by modamod** [reply]
FYI I found the issue with the db connection, I mounted a codespace folder to /var/lib/mysql which conflicted with the new config I was trying to apply. Once I removed the mount the connection worked as expected.

**Comment by ossanna16** [comment]
Hi there @modamod and welcome to our community! Thank you for asking a great question 🙂

**Reply by modamod** [reply]
Thank you. I appreciate the support :)

**Comment by michielbdejong** [comment]
Just as a heads-up, if your repo contains multiple dockerfiles (as it might well if you use docker-compose and you're building from source), an issue we ran into on the current Codespaces Beta is that you may have to run `touch Dockerfile` after the repo is checked out but before building any of them, see https://github.com/orgs/community/discussions/38878
